targets/zephyr/src/main-zephyr.c

```diff
@@ -158,6 +158,7 @@ const struct shell_cmd commands[] =
 void main (void)
 {
   printf ("Jerry Compilation " __DATE__ " " __TIME__ "\n");
+  jerry_init (JERRY_INIT_EMPTY);
   shell_register_app_cmd_handler (shell_cmd_handler);
   shell_init ("js> ", commands);
 } /* main */
```

Comment:
@LaszloLango

shell_init will spawn a fiber that performs the shell task.
task_fiber_start(stack, STACKSIZE, shell, 0, 0, 7, 0);

If you clean zephyr there it will just crash.

With the change of running code on eval i think pfalcon is right on not cleaning up after execution.